src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/SecretVolume.java

```diff
@@ -33,21 +33,34 @@
 import io.fabric8.kubernetes.api.model.VolumeBuilder;
 
 public class SecretVolume extends PodVolume {
+    private static final String DEFAULT_MODE = "420";
```

Comment:
why `420` ? 
the default in kubernetes is `0644` https://kubernetes.io/docs/concepts/configuration/secret/

I would set the default to empty string and then add it to the `buildVolume` if not empty

Reply:
Checkout this link under 'Secret files permissions' https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

Using JSON for kubernetes requests does not support Octal notation. 420 maps to 0644. Specifying octal notation in the YAML (without these changes) does not work, you have to use the JSON spec. I haven't dived too deep into the plugin but I am assuming that it uses JSON for requests.

Reply:
I have adjusted the code a bit. The new implementation gets rid of the defaults and only sets the defaultMode object if that variable is set.